pkg/porter/stamp.go

```diff
@@ -22,7 +22,10 @@ func (p *Porter) ensureLocalBundleIsUpToDate(opts bundleFileOptions) error {
 
 	if !upToDate {
 		fmt.Fprintln(p.Out, "Building bundle ===>")
-		return p.Build(BuildOptions{})
+		// opts.File is non-empty, which overrides opts.CNABFile if set
+		// (which may be if a cached bundle is fetched e.g. when running an action)
+		opts.CNABFile = ""
```

Comment:
I'm unclear on why clearing opts.CNABFile is necessary. Was this an existing bug? Something that came up after this PRs changes?

Reply:
Yes, I believe it was introduced now that we are passing in bundleFileOptions on the following line (previously, we called `p.Build()` without any).  We pass these in now so that when File is supplied on publish and an auto-rebuild is detected, that same File value migrates to the triggered build.